src/utils/EventEmitter.js

```diff
@@ -0,0 +1,77 @@
+// Source: https://gist.github.com/mudge/5830382?permalink_comment_id=2691957#gistcomment-2691957
+
+export default class EventEmitter {
+  constructor() {
+    /** @type {Events} */
+    this.events = {}
+  }
+
+  /**
+   * @param {string} eventName
+   * @returns {EventHandlers}
+   */
+  _getEventListByName(eventName) {
+    if (typeof this.events[eventName] === 'undefined') {
+      this.events[eventName] = new Set()
```

Comment:
A minor detail, but one at least to consider: `Set()` doesn't strictly guarantee any ordering. Most event registration systems typically imply a strict ordering (FIFO or LIFO). Not sure that people are going to care here, but...

if we wanted to slightly strengthen this design with a FIFO ordering guarantee, then:

1. `new Set()` could be changed o an `[]`
2. [the `.add(..)` (down on line 25)](https://github.com/sweetalert2/sweetalert2/blob/05d237aaa4924f03a7a239e68a3d8d71066931e3/src/utils/EventEmitter.js#L25) could be changed to:

    ```js
    const list = this._getEventListByName(eventName)
    if (!list.includes(eventHandler)) list.push(eventHandler)
    ```

Reply:
Addressed in https://github.com/sweetalert2/sweetalert2/pull/2763/commits/d061fd224f3e1f45792e201be027844afd33fac4